includes/class-filesystem-direct.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * The Class for WordPress Direct Filesystem with optimized read and write routines.
+ *
+ * @package aspire-update
+ */
+
+namespace AspireUpdate;
+
+/**
+ * The Class for WordPress Direct Filesystem with optimized read and write routines.
+ */
+class Filesystem_Direct extends \WP_Filesystem_Direct {
+
+	/**
+	 * Reads entire file into an array with options for limiting the number of lines and direction from the the lines are counted.
+	 *
+	 * @since 2.5.0
+	 *
+	 * @param string $file Path to the file.
+	 * @param int    $number_of_lines The number of lines to read. Default is -1 (read all lines).
+	 * @param bool   $count_bottom_to_up Count the lines from the bottom up. Default is false (count from top to bottom).
+	 *
+	 * @return array|false File contents in an array on success, false on failure.
+	 */
+	public function get_contents_array( $file, $number_of_lines = -1, $count_bottom_to_up = false ) {
+		if ( ! $this->exists( $file ) ) {
+			return false;
+		}
+
+		if ( -1 === $number_of_lines ) {
+			return @file( $file );
+		}
+
+		// phpcs:disable WordPress.WP.AlternativeFunctions.file_system_operations_fopen
+		/**
+		 * Extending WP_Filesystem methods for efficiency.  This is a valid use case.
+		 */
+		$handle = @fopen( $file, 'r' );
+		// phpcs:enable
```

Comment:
Let's just exclude this file from the following sniffs since they're always going to be valid in this file.

We can add this to the bottom of `phpcs.xml.dist`:

```xml
<!-- Exclude custom Filesystem API implementations from certain sniffs. -->
<rule ref="WordPress.WP.AlternativeFunctions.file_system_operations_fopen">
	<exclude-pattern>includes/class-filesystem-direct\.php</exclude-pattern>
</rule>
<rule ref="WordPress.WP.AlternativeFunctions.file_system_operations_fclose">
	<exclude-pattern>includes/class-filesystem-direct\.php</exclude-pattern>
</rule>
<rule ref="WordPress.WP.AlternativeFunctions.file_system_operations_fwrite">
	<exclude-pattern>includes/class-filesystem-direct\.php</exclude-pattern>
</rule>
<rule ref="WordPress.PHP.NoSilencedErrors.Discouraged">
	<exclude-pattern>includes/class-filesystem-direct\.php</exclude-pattern>
</rule>
```
We can then remove the inline comments for `phpcs:disable`, `phpcs:disable`, and the docblock-style comments explaining why it's used. I've included the `WordPress.PHP.NoSilencedErrors.Discouraged` sniff in the exclusions as that's also quite common with filesystem functions.